bluesky-tiled-plugins/bluesky_tiled_plugins/catalog_of_bluesky_runs.py

```diff
@@ -138,9 +138,24 @@ def v1(self):
             self._v1 = Broker(self)
         return self._v1
 
+    @staticmethod
+    def _namedtuple_asdict_default(obj):
+        """Convert NamedTuple to dict before serialization."""
+        if isinstance(obj, tuple) and hasattr(obj, "_asdict"):  # NamedTuple check
+            return obj._asdict()
```

Comment:
do dataclasses as well?

Reply:
Seemed like a great idea, but `orjson` handles data classes fine, but treats NamedTuples as ambiguous (is it an array or a dictionary).
Fun rabbit hole for the morning.